presto-orc/src/main/java/com/facebook/presto/orc/reader/MapSelectiveStreamReader.java

```diff
@@ -76,6 +88,7 @@ public void startStripe(Stripe stripe)
             currentReader = directReader;
         }
         else if (kind == DWRF_MAP_FLAT) {
+            checkState(flatReader != null, "flat map reader is null for stream %s", streamDescriptor);
```

Comment:
I'd suggest to throw OrcCorruptedException with something like "Flat map is not supposed to inside of a flat map"